R/create_project.R

```diff
@@ -0,0 +1,256 @@
+    "this project.",
+    "",
+    "## Getting Started",
+    "TODO: Guide users through getting your code up and running on their ",
+    "own system. ",
+    "In this section you can talk about:",
+    "1.	Installation process",
+    "2.	Software dependencies",
+    "3.	Latest releases",
+    "4.	API references",
+    "",
+    "# Build and Test",
+    "TODO: Describe and show how to build your code and run the tests.",
+    "",
+    "# Contribute",
+    "TODO: Explain how other users and developers can contribute to make ",
+    "your code better.",
+    "",
+    "## Git integration",
+    "If you want to use git with your project (you should!), ",
+    "please do the following steps (replace `<name of your repository>` with ",
+    "the actual name):",
+    "",
+    "1.  Go to your git repository provider (GitHub/Azure DevOps) and create ",
+    "a new repository",
+    "2.  DON'T check 'Add a README file'",
+    "3.  Go to the Terminal within RStudio and type the following commands ",
+    "(for the URL, e.g. https://github.com):",
+    "",
+    "```bash",
+    "git init",
+    "git branch -M main",
+    "git remote add origin <URL of your GitHub/Azure DevOps instance>/<name ",
+    "of your repository>.git",
+    "```",
+    "",
+    "4.  Restart RStudio",
+    "5.  Type in the R terminal `bash git add .` to add all files to ",
+    "the commit",
+    "6.  Type in the R terminal `bash git commit -m ",
+    "'Your commit message (initial commit)'` to commit those files with ",
+    "a message.",
+    "7.  In the terminal, execute the following command:",
+    "",
+    "```bash",
+    "git push -u origin main",
+    "```",
+    "",
+    "8.  For the following commits, repeat this process",
+    "",
+    "NOTE: For sharing content on GitHub you should have ticked the ",
+    "'Create a .gitignore file for GitHub' checkbox when creating the project.",
+    "This will give create a strict .gitignore which is suitable for sharing ",
+    "code to the public.",
+    "Please also review to ensure no sensitive information is shared.",
+    "",
+    "For more information about the integration of git and RStudio, ",
+    "check out https://happygitwithr.com."
+  )
+
+  # Write to README.md
+  readme_concat <- paste0(readme_content, collapse = "\n")
+  writeLines(readme_concat, con = file.path(path, "README.md"))
+
+
+  # Initialise renv
+  if (requireNamespace("renv", quietly = TRUE) && init_renv) {
+    renv::init(project = path,
+               bare = TRUE, load = FALSE, restart = FALSE)
+    renv::snapshot(project = path, prompt = FALSE,
+                   update = TRUE, packages = list("rmarkdown", "testthat",
+                                                  "renv", "stringr"))
+  } else {
+    warning(
+      paste0("renv couldn't be used as the `renv` package is not installed.",
+             "If you want to use renv, please first install it with `install",
+             ".packages('renv')`")
+    )
+  }
```

Comment:
It could be nice to add a message at the end of the function confirming the project has been made and telling the user to go to that directory?

Reply:
I like this idea. Have implemented with this:

```
cat(
    paste0("Your new dfeR project has been successfuly created! ",
           emo::ji("mortar_board"), "\n",
           "It exists at the file path: '", path, "'")
  )
```

May need to discuss nosiness of output as shows all the files created and the renv packages.